packages/expo-updates/ios/EXUpdates/Procedures/StartupProcedure.swift

```diff
@@ -121,8 +121,11 @@ final class StartupProcedure: StateMachineProcedure, AppLoaderTaskDelegate, AppL
       event = UpdatesStateEventCheckCompleteWithUpdate(manifest: manifest)
     case .rollBackToEmbedded(let commitTime):
       event = UpdatesStateEventCheckCompleteWithRollback(rollbackCommitTime: commitTime)
-    case .error(let error):
-      event = UpdatesStateEventCheckError(message: error.localizedDescription)
+    case .error:
```

Comment:
this is the change that fixes the invalid state transition. On android, we don't have the equivalent problem since we don't have an "error" remote check result (see implementation in `onRemoteCheckForUpdateFinished`). Maybe we should just remove the `.error` status? Or figure out why android doesn't have one.

Reply:
That is odd, I really thought Android had this case

Reply:
Update on this PR: I found the cause of the bug is that android and iOS were out of sync. I updated the PR to bring them into sync (removed the `.error` status since it's unnecessary). Also added a missing switch case to android.